app/forms/declaration/EntityDetails.scala

```diff
@@ -35,7 +35,8 @@ object EntityDetails {
       "eori" -> optional(
         text()
           .verifying("supplementary.eori.empty", nonEmpty)
-          .verifying("supplementary.eori.error", noLongerThan(17) and isAlphanumeric)
+          .verifying("supplementary.eori.nongb.error", isValidEORI)
+          .verifying("supplementary.eori.error", noLongerThan(17) and noShorterThan(14) and isAlphanumeric)
```

Comment:
Your validation `isValidEORI` checks if string starts from `GB/gb` and is followed by digits. Based on this in second validation `noLongerThan(17) and noShorterThan(14) and isAlphanumeric` it's enough for you to check the length of the input.

Checking `isAlphanumeric` don't give any value because you already validated it.

Reply:
Agreed..can remove that validation for isAlphanumeric

Reply:
Fixed